CONTRIBUTING.md

```diff
@@ -0,0 +1,109 @@
+
+## Code Guidelines
+
+To ensure consistency throughout the source code, keep these guidelines in mind as you are working:
+
+- We use [Prettier](https://prettier.io/) to automate our code formatting
+- TODO semicolons, etc.?
+- TODO editorconfig?, see https://github.com/gatsbyjs/gatsby/issues/3922
+
+## What is Gatsby?
+
+Gatsby lets you build a blazing-fast a static PWA (Progressive Web App) with your data—whatever the source—using React.js, Webpack, GraphQL, modern JavaScript, CSS, and more. Our [home page](https://www.gatsbyjs.org/) provides a high-level introduction as well as detailed information that we are continuously trying to improve:
+
+* [Documentation and Guides](https://www.gatsbyjs.org/docs/)
+  * [What are Guide articles?](https://www.gatsbyjs.org/docs/gatsby-style-guide/#what-are-guide-articles)
+* [Tutorial](https://www.gatsbyjs.org/tutorial/)
+* [Features](https://www.gatsbyjs.org/features/)
+* [Blog](https://www.gatsbyjs.org/blog/)
+* [Showcase](https://github.com/gatsbyjs/gatsby#showcase)
+
+## Development Setup
+
+Gatsby is built using Node.js and supports versions back to v4 and npm to v3. You also need [Git](https://git-scm.com/) and the [Gatsby CLI](https://www.npmjs.com/package/gatsby-cli).
+
+Please refer to our [Get started](https://www.gatsbyjs.org/docs/) section for detailed instructions how to get install and use Gatsby's command line tool.
+
+After cloning this repository or installing a "Gatsby Starter", run the following command to install the project dependencies:
+
+```bash
+npm install # or yarn
+```
+
+## What is a "Gatsby Starter"?
+
+Gatsby uses “starters” for starting new projects. Starters are partially built Gatsby sites pre-configured to help you get moving faster. There are several official starters and many others contributed from the Gatsby community! [See the Starters page for the full list](https://www.gatsbyjs.org/docs/gatsby-starters/)
+
+### Commonly used NPM scripts
+
+- **`npm develop`**: TODO
+- **`npm build`**: TODO
+- **`npm format`**: TODO
+
+## Project Structure
+
+- **`src`**: contains the source code. :-) TODO Explain subfolders, mention similar naming for these is not a must.
+  - **`components`**: TODO
+  - **`layouts`**: TODO
+  - **`pages`**: TODO
+  - **`templates`**: TODO
+  - **`utils`**: TODO
+- **`public`**: TODO
+- **`static`**: TODO
+
+[`gatsby-config.js`](https://www.gatsbyjs.org/docs/gatsby-config/) and the files exporting your [browser, Node, and SSR API](https://www.gatsbyjs.org/docs/gatsby-lifecycle-apis/) implementations
+
+* [`gatsby-browser.js`](https://www.gatsbyjs.org/docs/browser-apis/)
+* [`gatsby-node.js`](https://www.gatsbyjs.org/docs/node-apis/)
+* [`gatsby-ssr.js`](https://www.gatsbyjs.org/docs/ssr-apis/))
+
+live in the root directory of your project as siblings of the folders listed above.
+
+*Heads up*: TODO Explain why certain files mentioned here might not be in the starter that this document is part of.
+
+## Core Concepts
```

Comment:
I don't really understand this section very well--probably because I'm not a developer.